### ISSUE TITLE
plot of function with fractional exponent fails (version 1.4)

### ISSUE THREAD

**Issue by rolfdstoll** [sympy/sympy#17405]
Plotting a function with a broken exponent fails with sympy version 1.4. No error is displayed. The coordinate cross is displayed, but the graph of the function is missing.
With sympy version 1.3 the function is plotted correctly (see attached example).
System is Python 3.7.4 on Windows (WinPython)

![plot_test](https://user-images.githubusercontent.com/22221009/62953512-eea1f100-bded-11e9-93b4-9b7bb3117427.jpg)

**Comment by asmeurer** [comment]
Can you paste the code here so it is easier to copy-paste?

**Comment by rolfdstoll** [comment]
`import sympy as sp
from sympy import symbols
from sympy import init_printing
from sympy.plotting import plot

print(sp.__version__)
init_printing(use_unicode=True)
x, y = symbols('x y')
f= x**0.3-10*x**3+x**2
print(f)
p1 = plot(f, (x, -10, 10))`

**Comment by asmeurer** [comment]
Bisected to 

commit 1cf76c92cd4fed8070779982ac729b497939024b
Author: Jogi Miglani <jmig5776@gmail.com>
Date:   Mon Dec 10 00:09:03 2018 +0530

    fix for sympy.plot for logarithmic axis

CC @jmig5776

**Comment by rolfdstoll** [comment]
Sorry, I do not quite understand.
The changes shown in "fix for sympy.plot for logarithmic axis" are already part of the file      ...\ lib\site-packages\sympy\plotting\plot.py of my distribution.

**Comment by asmeurer** [comment]
That's the commit between 1.3 and 1.4 that broke this.

**Comment by jmig5776** [comment]
I am trying to debug this perhaps @oscargus may have some idea about this as he tried to fix issue like this on plot in #16796 .

**Comment by oscargus** [comment]
The reason seems to be that https://github.com/sympy/sympy/blob/e0f45aea9f283b1781a75e79ba47c93e46fc8bf6/sympy/plotting/plot.py#L665

returns `None` with negative starting points and fractional exponents. Plotting from 0 works well.

Not sure why because lambdifying the same function works and returns a complex value.

```
In [40]: f= x**0.3-10*x**3+x**2

In [41]: F=lambdify([x], f)

In [42]: F(-10.0)
Out[42]: (10101.172785763194+1.6142011210457226j)


In [46]: F?
Signature: F(x)
Docstring:
Created with lambdify. Signature:

func(x)

Expression:

x**0.3 - 10*x**3 + x**2

Source code:

def _lambdifygenerated(x):
    return (x**0.3 - 10*x**3 + x**2)


Imported modules:
File:      c:\users\oscar\sympy\sympy\<lambdifygenerated-5>
```

Not clear what the brackets do there... They are not seen on positive inputs.

Using rational numbers doesn't help, so it doesn't seem to be a floating-point issue.

One can see on the old plot that nothing is actually plotted for x < 0, so that may be a clue as well.

The problem indeed seems to be related to the bisected commit. There is a check for `None` here:
https://github.com/sympy/sympy/blob/e0f45aea9f283b1781a75e79ba47c93e46fc8bf6/sympy/plotting/plot.py#L630-L632
that causes the difference. I'll prepare a PR.

**Comment by oscargus** [comment]
Fix in #17409

**Comment by rolfdstoll** [comment]
Thanks, I applied the fix. Now it works.
